extensions/csv/src/main/java/io/deephaven/csv/containers/GrowableCharBuffer.java

```diff
@@ -0,0 +1,43 @@
+package io.deephaven.csv.containers;
+
+/**
+ * This is like TCharArrayList except that you can get at the underlying data buffer.
+ */
+public final class GrowableCharBuffer {
+    private static final int INITIAL_BUFFER_SIZE = 16384;
+
+    private char[] data = new char[INITIAL_BUFFER_SIZE];
+    private int size = 0;
+
+    public void append(char[] src, int srcOffset, int srcSize) {
+        ensure(srcSize);
+        System.arraycopy(src, srcOffset, data, size, srcSize);
+        size += srcSize;
+    }
+
+    private void ensure(int srcSize) {
+        final int sizeNeeded = Math.addExact(size, srcSize);
+        if (sizeNeeded <= data.length) {
+            return;
+        }
+
+        // Ensuring that we always at least double the buffer, but we may not always
+        // follow powers of two
+        final int newSize = Math.max(sizeNeeded, Math.multiplyExact(size, 2));
```

Comment:
Good point. If there were some library collection that would let me at its underlying data array, I would just use that.

I'm slightly reluctant to change because: this buffer is only used as a buffer for *cells*. That is the max size of this buffer is going to be the size of the largest cell in the input. So I think in all of the sane cases I can think of, this buffer will always be small. (10s or 100s of bytes at the most). Even giving it 16K as an initial size was probably way too generous.

Let me know what you think about the above argument.

Reply:
Ok, SG.  Maybe just add a line to the class javadoc to the effect that is used as a buffer for cells?